plugin/kubernetes/controller.go

```diff
@@ -395,3 +396,11 @@ func (dns *dnsControl) GetNodeByName(name string) (*api.Node, error) {
 	}
 	return v1node, nil
 }
+
+func (dns *dnsControl) GetNamespaceByName(name string) (*api.Namespace, error) {
+	v1ns, err := dns.client.Namespaces().Get(name, meta.GetOptions{})
+	if err != nil {
+		return &api.Namespace{}, err
```

Comment:
IMO, un-exposed namespaces should return NXDOMAIN... even though they exist, they are unexposed.

Reply:
But this check is not done here? Should it?

Reply:
Yes, before returning the namespace, verify that it is exposed.

Reply:
If you are leaving namespaces uncached, you can "cheat" here and skip the API lookup if only some namespaces are exposed.  I.e. just verify the namespace is on the exposed list without verifying it is real in the API. This means that if a user exposes a non-existing namespace, we'd present it as an empty namespace... but I think thats OK.

Reply:
Actually looking at the code, that is already checked before we go into findPods or findServices so I don't think we should do it here.
But some comments could clarify this.

Reply:
Should add namespace exposure cases unit tests.